sycl/source/detail/pi_opencl.cpp

```diff
@@ -78,6 +78,43 @@ pi_result OCL(piextDeviceSelectBinary)(
   return PI_SUCCESS;
 }
 
+pi_result OCL(piQueueCreate)(pi_context context, pi_device device,
+                             cl_command_queue_properties properties,
+                             pi_queue *queue) {
+  size_t devVerSize;
+  cl_int ret_err;
+  ret_err = clGetDeviceInfo(pi_cast<cl_device_id>(device), CL_DEVICE_VERSION, 0,
+                            NULL, &devVerSize);
+
+  if (ret_err != CL_SUCCESS)
+    return pi_cast<pi_result>(ret_err);
+
+  std::string devVer(devVerSize, '\0');
+  clGetDeviceInfo(pi_cast<cl_device_id>(device), CL_DEVICE_VERSION, devVerSize,
+                  &devVer.front(), NULL);
```

Comment:
I think it has to be front() in C++ < 17. non-const .data() for strings seems to be a C++17 addition: https://en.cppreference.com/w/cpp/string/basic_string/data

Reply:
Actually you are right, it does not work before C++17.
And since `devVer.front()` returns a reference according to the C++ specification, it works and is even shorten than `&*devVer.begin()`.